base_layer/core/src/mempool/unconfirmed_pool/unconfirmed_pool.rs

```diff
@@ -144,7 +144,9 @@ impl UnconfirmedPool {
                 .get(tx_key)
                 .ok_or_else(|| UnconfirmedPoolError::StorageOutofSync)?;
 
-            if curr_weight + ptx.weight <= total_weight {
+            if curr_weight + ptx.weight <= total_weight &&
+                !UnconfirmedPool::find_duplicate_input(&selected_txs, &ptx.transaction)
```

Comment:
I wonder if the duplicate check should not be implemented as a nested `IF` otherwise it will make the skip count system have funny behaviour that could result in partially filled blocks when there are many transactions with duplicate inputs.